src/components/Badge/Badge.scss

```diff
@@ -19,12 +18,14 @@ $pip-spacing: ($height - $pip-size) / 2;
   @include pip-color(var(--p-icon, color('ink', 'lightest')));
   display: inline-flex;
   align-items: center;
+  min-width: $height;
+  min-height: $height;
   padding: $vertical-padding $horizontal-padding;
   background-color: var(--p-surface-neutral, color('sky'));
   border: var(--p-override-zero, border-width(thick)) solid color('white');
   border-radius: $height;
   font-size: rem(13px);
-  line-height: $extra-small-height;
+  line-height: 1;
```

Comment:
Keep the line-height narrow and relative to the font-size helps avoid making the badge too tall, and also keeps it more compact in cases where the badge text itself wraps